spark/src/test/scala/org/apache/comet/exec/CometExecSuite.scala

```diff
@@ -62,6 +62,37 @@ class CometExecSuite extends CometTestBase {
     }
   }
 
+  test("CometShuffleExchangeExec logical link should be correct") {
+    withTempView("v") {
+      spark.sparkContext
+        .parallelize((1 to 4).map(i => TestData(i, i.toString)), 2)
+        .toDF("c1", "c2")
+        .createOrReplaceTempView("v")
+
+      withSQLConf(
+        SQLConf.ADAPTIVE_EXECUTION_ENABLED.key -> "false",
+        CometConf.COMET_COLUMNAR_SHUFFLE_ENABLED.key -> "true") {
+        val df = sql("SELECT * FROM v where c1 = 1 order by c1, c2")
+        val shuffle = find(df.queryExecution.executedPlan) {
+          case _: CometShuffleExchangeExec => true
+          case _ => false
+        }.get.asInstanceOf[CometShuffleExchangeExec]
+        assert(shuffle.logicalLink.isEmpty)
+      }
+
+      withSQLConf(
+        SQLConf.ADAPTIVE_EXECUTION_ENABLED.key -> "false",
+        CometConf.COMET_COLUMNAR_SHUFFLE_ENABLED.key -> "false") {
+        val df = sql("SELECT * FROM v where c1 = 1 order by c1, c2")
+        val shuffle = find(df.queryExecution.executedPlan) {
+          case _: ShuffleExchangeExec => true
+          case _ => false
+        }.get.asInstanceOf[ShuffleExchangeExec]
+        assert(shuffle.logicalLink.isEmpty)
```

Comment:
nit: these two clauses can be one in a loop to reduce the duplication

Reply:
Okay